presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/TimeUuidFunctions.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.cassandra;
+
+import com.datastax.driver.core.utils.UUIDs;
+import io.airlift.slice.Slice;
+import io.prestosql.spi.function.Description;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlType;
+
+import java.util.UUID;
+
+import static com.datastax.driver.core.utils.UUIDs.unixTimestamp;
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.prestosql.spi.type.DateTimeEncoding.packDateTimeWithZone;
+import static io.prestosql.spi.type.DateTimeEncoding.unpackMillisUtc;
+import static io.prestosql.spi.type.TimeZoneKey.UTC_KEY;
+
+public final class TimeUuidFunctions
+{
+    private TimeUuidFunctions() {}
+
+    @Description("Cassandra minTimeuuid function")
+    @ScalarFunction
+    @SqlType("varchar(36)")
+    public static Slice minTimeuuid(@SqlType("timestamp(3) with time zone") long timestamp)
```

Comment:
Why is this limited to `timestamp(3)` instead of supporting any precision?

Reply:
I assumed the value comes from Cassandra timestamp that is mapped to `timestamp(3)` in most cases, but I would change to any precision.